client/fingerprint/cpu.go

```diff
@@ -40,7 +40,9 @@ func (f *CPUFingerprint) Fingerprint(req *FingerprintRequest, resp *FingerprintR
 	}
 
 	if cfg.CpuCompute != 0 {
```

Comment:
At first glance this looks correct, but when I'm looking at the rest of `(*CPUFingerprint) Fingerprint` below it, I think the right fix is to remove this block entirely.

We check `if cfg.CpuCompute > 0` further down at [line 62](https://github.com/hashicorp/nomad/pull/9532/files#diff-b5ec91052e62adcb7b8376848d70311f75cfdc2237567384c6766a9ad5c0d0a9R64), but that's an unreachable condition because of this check here! And returning early from this block means we never set attributes like `cpu.modelname` or `cpu.numcores`.

Reply:
Thanks for the feedback.  I wasn't sure if omitting the other variables was intentional to avoid a mismatch of `cpu.numcores * cpu.frequency` and `cpu.totalcompute`, but I think your suggestion is the best implementation.

Updated the PR.